src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -39,13 +39,24 @@ object BitPat {
     * @param n the literal value as a string, in binary, prefixed with 'b'
     * @note legal characters are '0', '1', and '?', as well as '_' and white
     * space (which are ignored)
+    *
+    * @example {{{
+    * "0b10101".U === BitPat("b101??") // dynamically evaluates to true.B
+    * "0b10111".U === BitPat("b101??") // dynamically evaluates to true.B
+    * "0b10001".U === BitPat("b101??") // dynamically evaluates to false.B
```

Comment:
The leading zeros need to be removed, eg.`"b10101".U`. These examples cause runtime exceptions.